presto-main/src/main/java/com/facebook/presto/operator/scalar/MapDistinctFromOperator.java

```diff
@@ -68,14 +67,11 @@ public static boolean isDistinctFrom(
                 rightMapBlock,
                 (leftMapIndex, rightMapIndex) -> {
                     Object leftValue = readNativeValue(valueType, leftMapBlock, leftMapIndex);
-                    boolean leftNull = leftValue == null;
-                    if (leftNull) {
-                        leftValue = defaultValue(valueType.getJavaType());
-                    }
                     Object rightValue = readNativeValue(valueType, rightMapBlock, rightMapIndex);
+                    boolean leftNull = leftValue == null;
                     boolean rightNull = rightValue == null;
-                    if (rightNull) {
-                        rightValue = defaultValue(valueType.getJavaType());
+                    if (leftNull || rightNull) {
+                        return leftNull == rightNull;
```

Comment:
add a comment that nulls are not distinct from each other, but distinct from every other value

Reply:
we already use this property 21 lines earlier in the same (outer) method  (https://github.com/prestodb/presto/pull/10802/files#diff-61acc79c22fdf05c008fcdce7afda31aR55)